[TexMap] Add tests for damage propagation

Scony opened the pull request:
#### 17c6d1fb61adaa09480206696b7d721e0d21d4ca
<pre>
[TexMap] Add tests for damage propagation
<a href="https://bugs.webkit.org/show_bug.cgi?id=288465">https://bugs.webkit.org/show_bug.cgi?id=288465</a>

Reviewed by Nikolas Zimmermann.

This change adds layout tests covering the most important damage propagation scenarios.

* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/glib/damage/accelerated-canvas-2d-fillRect-expected.txt: Added.
* LayoutTests/platform/glib/damage/accelerated-canvas-2d-fillRect.html: Added.
* LayoutTests/platform/glib/damage/animations-blur-expected.txt: Added.
* LayoutTests/platform/glib/damage/animations-blur.html: Added.
* LayoutTests/platform/glib/damage/basic-propagation-001.html:
* LayoutTests/platform/glib/damage/canvas-webgl-expected.txt: Added.
* LayoutTests/platform/glib/damage/canvas-webgl.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/common.js:
(passTest):
* LayoutTests/platform/glib/damage/layer-downsize-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-downsize.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/layer-in-layer-movement-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-in-layer-movement.html: Added.
* LayoutTests/platform/glib/damage/layer-movement-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-movement.html: Added.
* LayoutTests/platform/glib/damage/layer-resize-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-resize.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/layer-upsize-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-upsize.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-arc-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-arc.html: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-downsizing-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-downsizing.html: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-fillRect-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-fillRect-w-offset-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-fillRect-w-offset.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-fillRect.html: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-new-canvas-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-new-canvas.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-path-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-path.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-putImageData-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-putImageData.html: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-resizing-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-resizing.html: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-text-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-text.html: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-upsizing-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-upsizing.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/scrolling-container-001-expected.txt: Added.
* LayoutTests/platform/glib/damage/scrolling-container-001.html: Added.
* LayoutTests/platform/glib/damage/scrolling-container-002-expected.txt: Added.
* LayoutTests/platform/glib/damage/scrolling-container-002.html: Added.
* LayoutTests/platform/glib/damage/scrolling-container-003-expected.txt: Added.
* LayoutTests/platform/glib/damage/scrolling-container-003.html: Added.
* LayoutTests/platform/glib/damage/scrolling-container-004-expected.txt: Added.
* LayoutTests/platform/glib/damage/scrolling-container-004.html: Added.
* LayoutTests/platform/glib/damage/scrolling-fullscreen-001-expected.txt: Added.
* LayoutTests/platform/glib/damage/scrolling-fullscreen-001.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/scrolling-fullscreen-002-expected.txt: Added.
* LayoutTests/platform/glib/damage/scrolling-fullscreen-002.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/scrolling-fullscreen-003-expected.txt: Added.
* LayoutTests/platform/glib/damage/scrolling-fullscreen-003.html: Added.
* LayoutTests/platform/glib/damage/transform-2d-rotate-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-2d-rotate.html: Added.
* LayoutTests/platform/glib/damage/transform-2d-scale-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-2d-scale.html: Added.
* LayoutTests/platform/glib/damage/transform-2d-skew-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-2d-skew.html: Added.
* LayoutTests/platform/glib/damage/transform-2d-translate-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-2d-translate.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-dynamic-addition-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-dynamic-addition.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-1-axis-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-1-axis.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-3-axes-content-update-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-3-axes-content-update.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-3-axes-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-3-axes.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-in-still-rotate-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-in-still-rotate.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-scale-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-scale.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-translate-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-translate-w-perspective-content-update-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-translate-w-perspective-content-update.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-translate-w-perspective-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-translate-w-perspective.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-translate.html: Added.
* LayoutTests/platform/wpe/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/291024@main">https://commits.webkit.org/291024@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1175bad02ecd6adb9df04df4a6359350d2e99f40

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/91775 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/11306 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/852 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/96731 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/42417 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/11685 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/19775 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/70460 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/27955 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/94776 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/8926 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/83147 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/50787 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/8676 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/739 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/41616 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/78987 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/740 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/98746 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/18919 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/13957 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/79491 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/19172 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/78991 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/78705 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19470 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/23227 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/557 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/11990 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/18907 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/18610 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/22065 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/20365 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

spenap commented on LayoutTests/platform/glib/damage/common.js:
```diff
@@ -40,10 +48,18 @@ function assert(condition, failureMessage) {
 }
 
+function assertEq(actual, expected, failureMessage) {
+    return assert(actual == expected, failureMessage + ", expected: " + expected + " but got: " + actual);
```
```suggestion
    return assert(actual == expected, `${failureMessage}, expected: ${expected} but got: ${actual}`);
```

spenap commented on LayoutTests/platform/glib/damage/common.js:
```diff
@@ -40,10 +48,18 @@ function assert(condition, failureMessage) {
+}
+
+function assertGt(actual, threshold, failureMessage) {
+    return assert(actual > threshold, failureMessage + ", " + actual + " is not greater than " + threshold);
```
```suggestion
    return assert(actual > threshold, `${failureMessage}, ${actual} is not greater than ${threshold}`);
```

spenap commented on LayoutTests/platform/glib/damage/common.js:
```diff
@@ -100,19 +121,40 @@ function log(entity) {
+    return newElement;
+}
+
 function createNewElementWithClass(elementName, className, lambda = (el) => {}) {
     var newElement = document.createElement(elementName);
     newElement.className = className;
     lambda(newElement);
     return newElement;
 }
```
```suggestion
function createNewElementWithClass(elementName, className, lambda = (el) => {}) {
    let decoratedLambda = (e, l) => { e.className = className; l(e); };
    return createNewElement(elementName, decoratedLambda);
}
```

I've tried the above locally on the WebInspector and it seems to work, which would allow us to make the `crewNewElementWithClass` and the `spawnNewElementWithClass` a specialisation of the `create` and `spawn` methods... which might not be a great gain. It made more sense when I first started thinking about it than when I see it now :-D

Scony replied:
let's keep it simple then

spenap commented on LayoutTests/platform/glib/damage/accelerated-canvas-2d-fillRect.html:
```diff
@@ -0,0 +1,39 @@
+      var canvas = document.getElementsByTagName("canvas")[0];
+      var ctx = canvas.getContext("2d");
+
+      processAnimationFrameSequence([
```
Most tests are skipping one RAF callback. We could make it part of the `processAnimationFrameSequence` behaviour, so it would skip when `callbackIndex == 0`.

Scony replied:
I've been thinking about it a lot and I think it's better not to have this kind of "magic" that would happen under the hood. IMO it's better to have that 1 line copy-pasted in all the tests as it makes the potential creators of new tests more aware.

Scony replied:
anyway, I've prepared more clean solution using the verbose argument passing mechanism

spenap commented on LayoutTests/platform/glib/damage/accelerated-canvas-2d-fillRect.html:
```diff
@@ -0,0 +1,39 @@
+              var damage = latestFrameDamage();
+              assertValid(damage);
+              // FIXME: Uncomment once the problems with full-screen and off-screen damage are fixed.
+              // assertRectsEq(damage.rects, [[0, 0, Math.min(window.innerWidth, 2000), Math.min(window.innerHeight, 2000)]]);
```
If different tests have sections that are commented out, we need to comb through them and restore these bits whenever we fix problems. Could it be possible to extract the functionality that doesn't yet work into its own test, and mark it as FAIL for now?

spenap commented on LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-new-canvas.html:
```diff
@@ -0,0 +1,27 @@
+          () => {
+              var damage = latestFrameDamage();
+              assertValid(damage);
+              assertRectsEq(damage.rects, [[0, 0, window.innerWidth, 54]]);
```
why 54 here?

Scony replied:
added a comment

spenap reviewed (commented):
This looks very, very thorough, thanks.

I've reviewed code, logic and typos, and concentrated more where there's actual behaviour. I haven't verified the test expectations: I feel I would need to run the test interactively and then double-check the rectangles, which would be time consuming. Would you like me to do that?

I've got two "generic" pieces of feedback:

1. FIXMEs: There were a few, and I think this might make things harder to maintain, as we would need to remember to go back, review tests and update them as the feature is improved and bugs are addressed them. If there's a way you could extract out the missing functionality into separate tests that fail, that'd be simpler
2. Cursor! When working on a different task, one test used a JavaScript-rendered cursor. As the cursor was moving around the screen, it left behind a halo slightly around its outer shape. I don't know if custom cursors use damage, and whether they could be tested here.

nikolaszimmermann reviewed (approved):
rs=me

Scony committed:
[TexMap] Add tests for damage propagation
https://bugs.webkit.org/show_bug.cgi?id=288465

Reviewed by Nikolas Zimmermann.

This change adds layout tests covering the most important damage propagation scenarios.

* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/glib/damage/accelerated-canvas-2d-fillRect-expected.txt: Added.
* LayoutTests/platform/glib/damage/accelerated-canvas-2d-fillRect.html: Added.
* LayoutTests/platform/glib/damage/animations-blur-expected.txt: Added.
* LayoutTests/platform/glib/damage/animations-blur.html: Added.
* LayoutTests/platform/glib/damage/basic-propagation-001.html:
* LayoutTests/platform/glib/damage/canvas-webgl-expected.txt: Added.
* LayoutTests/platform/glib/damage/canvas-webgl.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/common.js:
(passTest):
* LayoutTests/platform/glib/damage/layer-downsize-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-downsize.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/layer-in-layer-movement-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-in-layer-movement.html: Added.
* LayoutTests/platform/glib/damage/layer-movement-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-movement.html: Added.
* LayoutTests/platform/glib/damage/layer-resize-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-resize.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/layer-upsize-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-upsize.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-arc-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-arc.html: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-downsizing-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-downsizing.html: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-fillRect-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-fillRect-w-offset-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-fillRect-w-offset.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-fillRect.html: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-new-canvas-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-new-canvas.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-path-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-path.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-putImageData-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-putImageData.html: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-resizing-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-resizing.html: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-text-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-text.html: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-upsizing-expected.txt: Added.
* LayoutTests/platform/glib/damage/non-accelerated-canvas-2d-upsizing.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/scrolling-container-001-expected.txt: Added.
* LayoutTests/platform/glib/damage/scrolling-container-001.html: Added.
* LayoutTests/platform/glib/damage/scrolling-container-002-expected.txt: Added.
* LayoutTests/platform/glib/damage/scrolling-container-002.html: Added.
* LayoutTests/platform/glib/damage/scrolling-container-003-expected.txt: Added.
* LayoutTests/platform/glib/damage/scrolling-container-003.html: Added.
* LayoutTests/platform/glib/damage/scrolling-container-004-expected.txt: Added.
* LayoutTests/platform/glib/damage/scrolling-container-004.html: Added.
* LayoutTests/platform/glib/damage/scrolling-fullscreen-001-expected.txt: Added.
* LayoutTests/platform/glib/damage/scrolling-fullscreen-001.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/scrolling-fullscreen-002-expected.txt: Added.
* LayoutTests/platform/glib/damage/scrolling-fullscreen-002.html: Copied from LayoutTests/platform/glib/damage/basic-propagation-001.html.
* LayoutTests/platform/glib/damage/scrolling-fullscreen-003-expected.txt: Added.
* LayoutTests/platform/glib/damage/scrolling-fullscreen-003.html: Added.
* LayoutTests/platform/glib/damage/transform-2d-rotate-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-2d-rotate.html: Added.
* LayoutTests/platform/glib/damage/transform-2d-scale-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-2d-scale.html: Added.
* LayoutTests/platform/glib/damage/transform-2d-skew-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-2d-skew.html: Added.
* LayoutTests/platform/glib/damage/transform-2d-translate-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-2d-translate.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-dynamic-addition-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-dynamic-addition.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-1-axis-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-1-axis.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-3-axes-content-update-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-3-axes-content-update.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-3-axes-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-3-axes.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-in-still-rotate-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-rotate-in-still-rotate.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-scale-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-scale.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-translate-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-translate-w-perspective-content-update-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-translate-w-perspective-content-update.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-translate-w-perspective-expected.txt: Added.
* LayoutTests/platform/glib/damage/transform-3d-translate-w-perspective.html: Added.
* LayoutTests/platform/glib/damage/transform-3d-translate.html: Added.
* LayoutTests/platform/wpe/TestExpectations:

Canonical link: https://commits.webkit.org/291024@main
